src/ContainerLogs.jsx

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Terminal } from "xterm";
+
+import * as utils from './util.js';
+import varlink from './varlink.js';
+import cockpit from 'cockpit';
+
+import "./ContainerTerminal.css";
+
+const _ = cockpit.gettext;
+
+class ContainerLogs extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.onStreamClose = this.onStreamClose.bind(this);
+        this.onStreamMessage = this.onStreamMessage.bind(this);
+        this.connectStream = this.connectStream.bind(this);
+
+        const view = new Terminal({
+            cols: 80,
+            rows: 24,
+            convertEol: true,
+            cursorBlink: false,
+            disableStdin: true,
+            fontSize: 12,
+            fontFamily: 'Menlo, Monaco, Consolas, monospace',
+            screenReaderMode: true
+        });
+        view._core.cursorHidden = true;
+        view.write(_("Loading logs..."));
+
+        this.state = {
+            view: view,
+            opened: false,
+            loading: true,
+            errorMessage: "",
+            streamer: null,
+        };
+    }
+
+    componentDidMount() {
+        this._ismounted = true;
+        this.connectStream();
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevProps.width !== this.props.width) {
+            this.resize(this.props.width);
+        }
+    }
+
+    resize(width) {
+        var padding = 11 + 5 + 50;
```

Comment:
:) it would, but I just copied it from `src/ContainerTerminal.jsx` and even though I wrote that one as well, I don't fully remember how I came up with it :see_no_evil: